src/XMakeTasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1490,6 +1490,82 @@ Copyright (C) Microsoft Corporation. All rights reserved.
   </Target>
 
   <!--
+    ====================================================================================
+                                        _GetProjectReferenceTargetFrameworkProperties
+
+    Builds the GetTargetFrameworkProperties target of all existent project references,
+    passing $(TargetFrameworkMoniker) as $(ReferringTargetFramework) and sets the
+    SetTargetFramework metadata of the project reference to the value that is returned.
+
+    This allows a cross-targeting project to select how it should be configured to
+    build against the most appropriate target for the referring target framework.
+
+    ======================================================================================
+  -->
+  <Target Name="_GetProjectReferenceTargetFrameworkProperties" Outputs="%(_MSBuildProjectReferenceExistent.Identity)">
```

Comment:
> Or is it here to batch the target so the loose %(Identity) metadata below works?
Yes. This whole thing feels messy to me, but I couldn't find a better way to express it. I imagined there would be a nice way to say "invoke this task for each item and put an output value from the task in to metadata for the item", but I couldn't find one. Am I just missing the idiomatic way to do this?